io-sim/src/Control/Monad/IOSim/CommonTypes.hs

```diff
@@ -202,6 +204,12 @@ ppLabelled :: (a -> String) -> Labelled a -> String
 ppLabelled pp Labelled { l_labelled = a, l_label = Nothing  } = pp a
 ppLabelled pp Labelled { l_labelled = a, l_label = Just lbl } = concat ["Labelled ", pp a, " ", lbl]
 
+-- | Abstract unique symbols à la "Data.Unique".
+type role IOSimUnique nominal
+newtype IOSimUnique s = MkUnique{ unMkUnique :: Integer }
+  deriving stock   (Eq, Ord)
+  deriving newtype (Hashable, NFData)
+
```

Comment:
It's easier to read when data is defined first.
```suggestion
newtype IOSimUnique s = MkUnique{ unMkUnique :: Integer }
  deriving stock   (Eq, Ord)
  deriving newtype (Hashable, NFData)
type role IOSimUnique nominal
```

Reply:
Done.